src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -201,6 +202,9 @@ private IServiceCollection BuildHostingServices()
             // Conjure up a RequestServices
             services.AddTransient<IStartupFilter, AutoRequestServicesStartupFilter>();
 
+            // Ensure object pooling is available everywhere.
+            services.TryAddSingleton<ObjectPoolProvider, DefaultObjectPoolProvider>();
```

Comment:
Why _not_ allow users to override the provider? In any case, this method uses both Try and unconditional; I went with Try.